cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```
@@ -126,6 +127,39 @@ public void deleteRepository(FullHttpRequest request, HttpResponder responder,
                                                               namespaceId));
   }
 
+  /**
+   * Lists applications found in linked repository for the given namespace.
+   * Returns the application name and git-filehash for each application config file found
+   * <pre>
+   * {@code
+   * {
+   *  apps:[
+   *    {
+   *      "name": "xyz",
+   *      "fileHash": "abc"
+   *    }
+   *  ]
+   * }
+   * }
+   *
+   * </pre>
+   */
+  @GET
+  @Path("/apps")
+  public void listApplications(FullHttpRequest request, HttpResponder responder,
+                               @PathParam("namespace-id") String namespaceId) throws Exception {
+    checkSourceControlFeatureFlag();
+    NamespaceId namespace = validateNamespaceId(namespaceId);
+    try {
+      ListAppsResponse listResponse = sourceControlService.listApps(namespace);
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(listResponse));
+    } catch (AuthenticationConfigException e) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, e.getMessage());
```

looks like this exception is thrown when we can't get the password from the secure store. 400 is normally used when the request itself is wrong (client error), which means a retry would never succeed. A SecureStore issue does not fall under this category.

Unfortunately the SecureStore API is bad, as there is no way to differentiate between the key not existing or some other problem. Ideally we would return a different code if the key doesn't exist (403? 404?) and a 500 if we can't reach the service for some reason. Without a way to differentiate, I want to say we should make this a 500, which means you can just let the exception propagate up instead of catching and handling.